src/math.jl

```diff
@@ -55,6 +55,15 @@ end
 Base.:*(l::AbstractDimensions, r::AbstractDimensions) = map_dimensions(+, l, r)
 Base.:/(l::AbstractDimensions, r::AbstractDimensions) = map_dimensions(-, l, r)
 
+# Multiplying ranges with units
+for (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.:*(r::StepRange, q::$type) = (first(r) * q):(Base.step(r) * q):(last(r) * q)
```

Comment:
This does not seem safe from floating point weirdness. 

I would call `range(start, stop, length)` for all cases.

Reply:
Good call!